### DISCUSSION TITLE
Strange class name in ValueWrapper.php

### DISCUSSION THREAD

**Discussion by albert-github** [symfony/symfony#46477] [Q&A] [answered]
Is there a special reason why there is in the file: src/Symfony/Component/Cache/Traits/ValueWrapper.php the ISO-8859-1 encoded line:
```
class ©
```
and not a bit a more readable or at least an UTF-8 encoded name?
(as far as I can see the file is new in Symfony 6.1.0).

**Comment by nicolas-grekas** [comment, marked as answer]
It's a "cookie" class, one with a unique distinct super-short name, internal also, needed for technical reasons (namely minimizing the size of the serialized payload).
I made it non-utf8 on purpose, to have a single-char class-name that has low collision chance.

**Reply by albert-github** [reply]
You wrote:
> I made it non-utf8 on purpose, to have a single-char class-name that has low collision chance.

But won't it give strange effects when something else is UTF-8?

**Reply by nicolas-grekas** [reply]
I don't anticipate any issues related to that :shrug: 
Did you experience any?

**Reply by albert-github** [reply]
No I didn't experience any so far, but from a theoretical point of view / Murphy I can imagine that something can go wrong in the future.

**Reply by albert-github** [reply]
As a side note you are now using multiple encodings as in e.g. ./src/Symfony/Component/Validator/Constraints/Isin.php (i.e. it is in a comment though) there is the name Masforné in UTF-8 encoding.

**Reply by DocFX** [reply]
Unearthing this: for some reason, after doing the composer require stuff to get the cache on Windows 11 / NTFS, files get to be using UTF-8 judging from editor, but after the clone, I always get `"Cannot find class �"`, that's looking for `src/Symfony/Component/Cache/Traits/ValueWrapper.php`.

Only solution I found on both project is to trash the `vendor` and `var/cache` dir and do a `composer update`.

The Unicode `©` becomes `�`. It remains so, but the project then works. I still have no clue where this comes from. I was thinking about the BOM on files, ot some extremely weird stuff related to Composer itself.

Note: this was after requiring `twig/cache-extra`.

Just letting this comment in case someone else stumbles on this.

**Reply by madmis** [reply]
> Did you experience any?

I've experienced. 
I try to profiling my code with Xhprof and it's report is not readable ([by xhprof ui](https://github.com/longxinH/xhprof)) because of this non utf8 symbols. 

Also i try to use [Buggregator](https://github.com/buggregator/) and it's require json encoded report. But it's impossible to json encode (`error: Malformed UTF-8 characters, possibly incorrectly encoded`) values with this symbols. 

Seems like this feature makes it impossible to profiling Symfony application when using Symfony cache.
That's very unfortunate. It means I'll have to write a custom cache.

<img width="1238" height="119" alt="image" src="https://github.com/user-attachments/assets/b3b20d30-b63b-4e37-8b07-88bcf95bcf8b" />

Could you suggest something, @nicolas-grekas?
Maybe these changes should be rolled back?

**Reply by nicolas-grekas** [reply]
That's an issue with both tools, nothing forces using UTF-8 for class names.

**Comment by staabm** [comment]
its not something to be fixed on the symfony side, but just to give you an idea: the autoloader in the REDAXO project uses a json cache file for the autoloader, which does not work with non-UTF8 stuff. I try to fix it at the REDAXO end though.

see german issue: https://github.com/redaxo/redaxo/issues/5780

**Comment by restena-sw** [comment]
By using such a character, you deliberately break PSR-1. https://www.php-fig.org/psr/psr-1/ 

```Files MUST use only UTF-8 without BOM for PHP code.```

I don't think it is very unreasonable that other tools expect to see UTF-8 everywhere in 2025. 

It is one option to state "all those other tools are at fault, they should be generalised enough to support all kinds of charsets." Another option would be to stop using the character and do UTF-8 like everyone else.
